app/services/creators/other_incomes_creator.rb

```diff
@@ -6,15 +6,19 @@ class OtherIncomesCreator < BaseCreator
 
     attr_reader :other_income_sources
 
-    def initialize(assessment_id:, other_incomes: [])
+    def initialize(assessment_id:, other_incomes_params:)
       super()
       @assessment_id = assessment_id
-      @other_incomes = other_incomes[:other_incomes]
+      @other_incomes_params = other_incomes_params
       @other_income_sources = []
     end
 
     def call
-      create_records
+      if json_validator.valid?
+        create_records
+      else
+        self.errors = json_validator.errors
```

Comment:
not a blocker/for discussion I have changed this in my APIPIE to below as errors method is instantiated to an empty array and ideally errors should be pushed/added to it rather than overwriting the entire array (which would make the code more breakable if something were to add an error first then this overwrites)
```suggestion
         errors.concat(json_validator.errors)
```

Reply:
Thanks 👍 have made suggested change